app/pages/patientdata/patientdata.js

```diff
@@ -526,6 +514,62 @@ export let PatientData = React.createClass({
       datetimeLocation: datetime
     });
   },
+
+  componentWillMount: function() {
+    this.doFetching(this.props);
+    this.doProcessing(this.props);
+    var params = this.props.queryParams;
+
+    if (!_.isEmpty(params)) {
+      var prefs = _.cloneDeep(this.state.chartPrefs);
+      prefs.bolusRatio = params.dynamicCarbs ? 0.5 : 0.35;
+      prefs.dynamicCarbs = params.dynamicCarbs;
+      this.setState({
+        chartPrefs: prefs
+      });
+    }
+  },
+
+  componentWillUnmount: function() {
+    this.props.clearPatientData(this.props.patient.userid);
+  },
+
+  componentWillReceiveProps: function(nextProps) {
+    var userId = this.props.routeParams.id;
+    var currentPatientData = _.get(this.props, ['patientDataMap', userId], null);
+    
+    var nextPatientData = _.get(nextProps, ['patientDataMap', userId], null);
+
+    if (!currentPatientData && nextPatientData) {
+      this.doProcessing(nextProps);
+    }
+  },
+
+  doProcessing: function(nextProps) {
+    // I realised the one source of truth we can rely on here to ensure we show the correct patient
```

Comment:
I think I quite like this idea. Will want to ruminate on it and discuss more as more of the state reorg happens - basically will want to make sure that the route is then the single source of truth for 'patient in view' full stop (app-wide), but I think that makes a lot of sense and puts the router in control in a way that feels quite right to me, actually.